Documentation (#3)
vtex-apps/seller-selector: docs/README.md

> # VTEX Seller Selector

Reviewer: ```suggestion
# Seller Selector
```

> ![exampleLayout](https://user-images.githubusercontent.com/53904010/78715148-595eea00-78f3-11ea-90fa-dc38c37a80d8.png)

Reviewer: ```suggestion
```

Author: Already fixed.

> Seller Selector enables your visitors to easily see how many sellers your marketplace has for certain products, check their prices
> and add then to their cart.

Reviewer: ```suggestion
Seller Selector displays the number of sellers your marketplace has for each product. It enables users to compare prices from each seller and add the desired product to their cart.

![exampleLayout](https://user-images.githubusercontent.com/53904010/78715148-595eea00-78f3-11ea-90fa-dc38c37a80d8.png)

```

> ## Configuration:

Reviewer: ```suggestion
## Configuration
```

> 1. Import the Seller Selector app to your theme's dependencies in the `manifest.json` as shown below:

Reviewer: ```suggestion
1. Add the Seller Selector app to your theme's dependencies in the `manifest.json` as shown below:
```

```json
"dependencies": {
  "vtex.sellerselector": "0.x"
}
```

> 2. Add the `link-seller` block inside your product page, this will enable a link to the Seller Selector main page, for example:

Reviewer: ```suggestion
2. Add the `link-seller` block, exported by the `vtex.sellerselector` app, inside your product page (`store.product` template). When rendered, the block will create a link to the Seller Selector main page. For example:
```

> ```json
> "flex-layout.col#right-col": {
>     "props": {
>       "preventVerticalStretch": true,
>       "rowGap": 0
>     },
>     "children": [
>       "product-name",
>       "link-seller"
>       "product-price#product-details",
>       "product-separator",
>       "flex-layout.row#buy-button",
>       "availability-subscriber",
>       "shipping-simulator",
>       "share#default"
>     ]
>   },
> ```

Reviewer: ```suggestion
```diff
"flex-layout.col#right-col": {
    "props": {
      "preventVerticalStretch": true,
      "rowGap": 0
    },
    "children": [
      "product-name",
+     "link-seller"
      "product-price#product-details",
      "product-separator",
      "flex-layout.row#buy-button",
      "availability-subscriber",
      "shipping-simulator",
      "share#default"
    ]
  },
```
```

> 3. The page cointaining all sellers will be displayed automatically, but you can configure the page layout and props for each block.

Reviewer: ```suggestion

By declaring the `link-seller` block, a page containing all available sellers will be displayed automatically. However, you can configure the Seller Selector page layout, using props for each one of the blocks used to build it behind the scenes. 

```

# Advanced Configuration

> According to the Seller Selector composition, it can be highly customizable using it's specific blocks. Currently, its default implementation is as follows:

Reviewer: ```suggestion
Behind the scenes, the Seller Selector page uses the following default implementation:
```

```json
{
  "store.sellers": {
    "blocks": [
      "seller-table"
    ]
  },
  "seller-table": {
    "children": [
      "vtex.store-components:product-name",
      "seller-simulate-shipping",
      "seller-head",
      "seller-body"
    ]
  },
  "seller-head": {
    "children": [
      "seller-head-cell#Seller",
      "seller-head-cell#Price",
      "seller-head-cell#Shipping",
      "seller-head-cell#PriceWithShipping",
      "seller-head-cell#BuyButton"
    ]
  },
  "seller-head-cell#Seller": {
    "props": {
      "title": "Seller"
    }
  },
  "seller-head-cell#Price": {
    "props": {
      "title": "Price"
    }
  },
  "seller-head-cell#Shipping": {
    "props": {
      "title": "Shipping"
    }
  },
  "seller-head-cell#PriceWithShipping": {
    "props": {
      "title": "Price With Shipping"
    }
  },
  "seller-head-cell#BuyButton": {
    "props": {
      "title": "Add To Cart"
    }
  },
  "seller-body": {
    "children": [
      "seller-row"
    ]
  },
  "seller-row": {
    "children": [
      "seller-name",
      "seller-price",
      "seller-shipping",
      "seller-price-with-shipping",
      "seller-add-to-cart"
    ]
  }
}
```

> # Props Configuration

Reviewer: ```suggestion
After presenting the default implementation, we should have a table with all blocks used to build the page and a description to clarify their functionality. You can use the Search Results documentation (https://vtex.io/docs/components/all/vtex.search-result@3.54.0/) as an example.  

Also, we have a bunch of blocks with no props. Seller Name, Seller Price, Seller Shipping. Could we have more details about them?
```

Author: I've added some descriptions for each block.

> #### seller-table
> 
> | Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
> | ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
> | `limitShownShippingInformation`       | `Number`  | Max number of shipping options to be shown.                                                                                                                                                                             | `3`         |

Reviewer: ```suggestion
### `seller-table` props

| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `limitShownShippingInformation`       | `number`  | Max number of shipping options to be displayed.                                                                                                                                                                             | `3`         |
```

> #### seller-add-to-cart
> 
> | Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
> | ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
> | `isOneClickBuy`          | `Boolean` | Whether the button button add the product to the cart and stay in the page or not                                                                                                                        | false         |

Reviewer: ```suggestion
### `seller-add-to-cart` props

| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `isOneClickBuy`          | `boolean` | Defines if users will keep navigating in the same page once the Buy button was clicked on (`true`) or if they will be redirected (`false`)   | `false`         |
```

Reviewer: Which page will users be redirected for? (considering that the `false`  value was configured)

Author: The user will be redirected for the cart page, for him to follow the checkout flow. I'll commit your suggestion and add this information too!

> #### seller-head-cell

Reviewer: ```suggestion
### `seller-head-cell` props
```

> | Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
> | ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
> | `title`     | `String`  | Text to display on each table header cell                                                                                                                

Reviewer: ```suggestion
| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `title`     | `string`  | Text displayed on each table header cell. | `undefined`|                                                                                                                 
```

Reviewer: What is a table header cell?

```json
"dependencies": {
  "vtex.sellerselector": "0.x"
}
```

2. Add the `link-seller` block inside your product page, this will enable a link to the Seller Selector main page, for example:

```json
"flex-layout.col#right-col": {
    "props": {
      "preventVerticalStretch": true,
      "rowGap": 0
    },
    "children": [
      "product-name",
      "link-seller"
      "product-price#product-details",
      "product-separator",
      "flex-layout.row#buy-button",
      "availability-subscriber",
      "shipping-simulator",
      "share#default"
    ]
  },
```
3. The page cointaining all sellers will be displayed automatically, but you can configure the page layout and props for each block.

# Advanced Configuration
According to the Seller Selector composition, it can be highly customizable using it's specific blocks. Currently, its default implementation is as follows:

```json
 {
  "store.sellers": {
    "blocks": [
      "seller-table"
    ]
  },
  "seller-table": {
    "children": [
      "vtex.store-components:product-name",
      "seller-simulate-shipping",
      "seller-head",
      "seller-body"
    ],
    "props": {

>       "limitShownShippingInformation": 5

Reviewer 2: Não precisa disso aqui no default, já existe um valor default (3). Foi colocado só para teste

Author: Fixed

    }
  },
  "seller-head": {
    "children": [
      "seller-head-cell#Seller",
      "seller-head-cell#Price",
      "seller-head-cell#Shipping",
      "seller-head-cell#PriceWithShipping",
      "seller-head-cell#BuyButton"
    ]
  },
  "seller-head-cell#Seller": {
    "props": {
      "title": "Seller"
    }
  },
  "seller-head-cell#Price": {
    "props": {
      "title": "Price"
    }
  },
  "seller-head-cell#Shipping": {
    "props": {
      "title": "Shipping"
    }
  },
  "seller-head-cell#PriceWithShipping": {
    "props": {
      "title": "Price With Shipping"
    }
  },
  "seller-head-cell#BuyButton": {
    "props": {
      "title": "Add To Cart"
    }
  },
  "seller-body": {
    "children": [
      "seller-row"
    ]
  },
  "seller-row": {
    "children": [
      "seller-name",
      "seller-price",
      "seller-shipping",
      "seller-price-with-shipping",
      "seller-add-to-cart"
    ]
  },
  "seller-add-to-cart": {
    "props": {

>       "OneClickBuy": true

Reviewer 2: Mesma coisa, foi incluído pra teste e não precisa no default

Author: Fixed

# Seller Selector

> Seller Selector displays the number of sellers your marketplace has for each product. It enables users to compare prices from each seller and add the desired product to their cart.

Reviewer: ```suggestion
Seller Selector displays the number of [**sellers**](https://help.vtex.com/tutorial/what-is-a-seller--5FkLvhZ3Few4CWWIuYOK2w) your marketplace has for each product. It enables users to compare prices from each seller and add the desired product to their cart.
```

![exampleLayout](https://user-images.githubusercontent.com/53904010/78715148-595eea00-78f3-11ea-90fa-dc38c37a80d8.png)

## Configuration

1. Add the Seller Selector app to your theme's dependencies in the `manifest.json` as shown below:

```json
"dependencies": {
  "vtex.sellerselector": "0.x"
}
```

2. Add the `link-seller` block, exported by the `vtex.sellerselector` app, inside your product page (`store.product` template). When rendered, the block will create a link to the Seller Selector main page. For example:

```diff
"flex-layout.col#right-col": {
    "props": {
      "preventVerticalStretch": true,
      "rowGap": 0
    },
    "children": [
      "product-name",
+     "link-seller"
      "product-price#product-details",
      "product-separator",
      "flex-layout.row#buy-button",
      "availability-subscriber",
      "shipping-simulator",
      "share#default"
    ]
  },
```

By declaring the `link-seller` block, a page containing all available sellers will be displayed automatically. However, you can configure the Seller Selector page layout, using props for each one of the blocks used to build it behind the scenes. 

> # Advanced Configuration

Reviewer: ```suggestion
### Advanced Configuration
```

Behind the scenes, the Seller Selector page uses the following default implementation:

> ```diff

Reviewer: ```suggestion
```json
```

{
  "store.sellers": {
    "blocks": [
      "seller-table"
    ]
  },
  "seller-table": {
    "children": [
      "vtex.store-components:product-name",
      "seller-simulate-shipping",
      "seller-head",
      "seller-body"
    ]
  },
  "seller-head": {
    "children": [
      "seller-head-cell#Seller",
      "seller-head-cell#Price",
      "seller-head-cell#Shipping",
      "seller-head-cell#PriceWithShipping",
      "seller-head-cell#BuyButton"
    ]
  },
  "seller-head-cell#Seller": {
    "props": {
      "title": "Seller"
    }
  },
  "seller-head-cell#Price": {
    "props": {
      "title": "Price"
    }
  },
  "seller-head-cell#Shipping": {
    "props": {
      "title": "Shipping"
    }
  },
  "seller-head-cell#PriceWithShipping": {
    "props": {
      "title": "Price With Shipping"
    }
  },
  "seller-head-cell#BuyButton": {
    "props": {
      "title": "Add To Cart"
    }
  },
  "seller-body": {
    "children": [
      "seller-row"
    ]
  },
  "seller-row": {
    "children": [
      "seller-name",
      "seller-price",
      "seller-shipping",
      "seller-price-with-shipping",
      "seller-add-to-cart"
    ]
  }
}
```

> Now, you are able to build your own page listing the sellers of a product, you can check below the description for each block:

Reviewer: ```suggestion

```

> | Block name   | Description  |
> | -------- | ------------------------ |
> | `seller-table`     |  Layout block that enables a table for you to build the seller-selector page using it's composition blocks. It's possible to use three main blocks `seller-simulate-shipping`, `seller-head`, `seller-body` inside this one.
> | `seller-simulate-shipping`  | Block that builds a form for the user to add their postal code, and simulate the shipping costs to his address |
> | `seller-head`  | Builds a header to be used on the seller-selector table. You can use `seller-head-cell` inside this block|
> | `seller-head-cell` | Used to define each title per column inside the table header. |
> | `seller-body`  | Builds a body to load all the sellers info inside the table |
> | `seller-row`  | Set a division to be used inside the seller table, each block inside it creates a new column to be loade. You can use `seller-name`, `seller-price`, `seller-shipping`, `seller-price-with-shipping`, `seller-add-to-cart` inside this block |
> | `seller-name` | Displays the seller name for each seller connected to the specific product context |
> | `seller-price` | Displays the seller price for each seller connected to the specific product context |
> | `seller-shipping` | Shipping cost considering the seller informations |
> | `seller-price-with-shipping` | Shipping cost plus the product price for the user to understand the purchase final cost |
> | `seller-add-to-cart` | Builds a buy button that adds to the store cart the product connected with the specific row seller  |

Reviewer: ```suggestion
| Block name   | Description  |
| -------- | ------------------------ |
| `seller-table`     |  Layout block that provides a table to build the Seller Selector page with other blocks. It is possible to build the page using three main blocks: `seller-simulate-shipping`, `seller-head` and `seller-body` (declared as children of `seller-head`). 
| `seller-simulate-shipping`  | Builds a form so users can add a postal code and then simulate the shipping costs to the desired address. |
| `seller-head`  | Builds a header to be used on the Seller Selector table. You can pass to it the `seller-head-cell` block as children. |
| `seller-head-cell` | Defines a title for each column in the table header. |
| `seller-body`  | Defines the page main content. It is responsible for displaying all sellers' data in the table body.|
| `seller-row`  | Used inside the Seller Selector table to separate seller's data into columns. You can use the blocks listed below (`seller-name`, `seller-price`, `seller-shipping`, `seller-price-with-shipping`, `seller-add-to-cart`) as `seller-row`'s children in order to provide all needed seller's data. |
| `seller-name` | Displays the seller name. |
| `seller-price` | Displays the seller price for a given product. |
| `seller-shipping` | Displays shipping cost considering the sellers' data. |
| `seller-price-with-shipping` | Displays the purchase final cost (shipping cost + product price). |
| `seller-add-to-cart` | Displays a Buy button that adds a given seller's product to the shopping cart.  |
```

> ### `seller-table` props
> 
> | Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
> | ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
> | `limitShownShippingInformation`       | `number`  | Max number of shipping options to be displayed.                                                                                                                                                                             | `3`         |
> 

Reviewer: ```suggestion
#### `seller-table` props

| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `limitShownShippingInformation`       | `number`  | Max number of shipping options to be displayed.                                                                                                                                                                             | `3`         |

```

> ### `seller-add-to-cart` props
> 
> | Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
> | ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
> | `isOneClickBuy`          | `boolean` | Defines if users will keep navigating in the same page once the Buy button was clicked on (`true`) or if they will be redirected (`false`)   | `false`         |

Reviewer: ```suggestion
#### `seller-add-to-cart` props

| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `isOneClickBuy`          | `boolean` | Defines if users will keep navigating in the same page once the Buy button was clicked on (`true`) or if they will be redirected (`false`)   | `false`         |
```

> ### `seller-head-cell` props
> 
> | Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
> | ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
> | `title`     | `string`  | Text displayed on each table header cell. | `undefined`|                                                                                                                 

Reviewer: ```suggestion
#### `seller-head-cell` props

| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `title`     | `string`  | Text displayed on the table header for each column. | `undefined`|                                                                                                                 
```
